mailpoet/tests/integration/Segments/DynamicSegments/Filters/WooCommerceSubscriptionTest.php

```diff
@@ -0,0 +1,253 @@
+
+  public function testAllSubscribersFoundWithOperatorNoneOf() {
+    $product = $this->createProduct("Another newsletter");
+    $notToBeFoundEmail = "not-to-be-found@example.com";
+    $subscriberId = $this->tester->createWordPressUser($notToBeFoundEmail, "subscriber");
+    $this->assertTrue(!is_wp_error($subscriberId), "User could not be created $notToBeFoundEmail");
+
+    $this->createSubscription(
+      [],
+      (int)$subscriberId,
+      $product
+    );
+    $testee = $this->diContainer->get(WooCommerceSubscription::class);
+    $queryBuilder = $this->getQueryBuilder();
+    $filter = $this->getSegmentFilter(
+      DynamicSegmentFilterData::OPERATOR_NONE,
+      [$product]
+    );
+
+
+    $resultQuery = $testee->apply($queryBuilder, $filter);
+
+    $foundSubscribers = $this->getEmailsFromQueryBuilder($resultQuery);
+
+    $this->assertCount(3, $foundSubscribers);
+    $this->assertFalse(in_array($notToBeFoundEmail, $foundSubscribers));
+    $this->tester->deleteWordPressUser($notToBeFoundEmail);
+  }
+
+  public function testAllSubscribersFoundWithOperatorAllOf() {
+    $this->createProduct("Another newsletter");
+    $notToBeFoundEmail = "not-to-be-found@example.com";
+    $toBeFoundEmail = "find-me@example.com";
+    $this->tester->deleteWordPressUser($toBeFoundEmail);
+    $this->tester->deleteWordPressUser($notToBeFoundEmail);
+    $notToBeFoundSubscriberId = $this->tester->createWordPressUser($notToBeFoundEmail, "subscriber");
+    $toBeFoundSubscriberId = $this->tester->createWordPressUser($toBeFoundEmail, "subscriber");
+    $this->assertTrue(!is_wp_error($toBeFoundSubscriberId), "Could not create user $toBeFoundEmail");
+    $this->assertTrue(!is_wp_error($notToBeFoundSubscriberId), "Could not create user $notToBeFoundEmail");
+
+    $this->createSubscription(
+      [],
+      (int)$toBeFoundSubscriberId,
+      ...$this->products
+    );
+    $testee = $this->diContainer->get(WooCommerceSubscription::class);
+    $queryBuilder = $this->getQueryBuilder();
+    $filter = $this->getSegmentFilter(
+      DynamicSegmentFilterData::OPERATOR_ALL,
+      $this->products
+    );
+
+
+    $sql = sprintf(
+      "SELECT DISTINCT wp_mailpoet_subscribers.id FROM wp_mailpoet_subscribers INNER JOIN wp_postmeta postmeta ON postmeta.meta_key = '_customer_user' AND wp_mailpoet_subscribers.wp_user_id=postmeta.meta_value INNER JOIN wp_posts posts ON postmeta.post_id = posts.id AND posts.post_type = 'shop_subscription' AND posts.post_status IN('wc-active', 'wc-pending-cancel') INNER JOIN wp_woocommerce_order_items items ON postmeta.post_id = items.order_id AND order_item_type = 'line_item' INNER JOIN wp_woocommerce_order_itemmeta itemmeta ON itemmeta.order_item_id=items.order_item_id AND itemmeta.meta_key='_product_id' WHERE itemmeta.meta_value IN (%s) GROUP BY wp_mailpoet_subscribers.id HAVING COUNT(wp_mailpoet_subscribers.id) = 2",
+      implode(',', $this->products)
+    );
+
+    global $wpdb;
+    $result = $wpdb->get_results($sql);
```

Comment:
I see that this doesn't create any data in the DB and also the $result variable is not used. Isn't this some debug leftover or so?

Reply:
Debug leftover... Thx!